Client/Frontend/UserContent/UserScripts/FarblingProtection.js

```diff
@@ -0,0 +1,304 @@
+      },
+      length: {
+        get: function () {
+          return pluginData.mimeTypes.length
+        }
+      }
+    })
+
+    // Create mime-types and link them to the new plugin
+    for (const [index, mimeType] of pluginData.mimeTypes.entries()) {
+      const newMimeType = makeFakeMimeType(mimeType)
+
+      newPlugin[index] = newMimeType
+      newPlugin[newMimeType.type] = newMimeType
+
+      Reflect.defineProperty(newMimeType, 'enabledPlugin', {
+        get: function () {
+          return newPlugin
+        }
+      })
+    }
+
+    // Fix .item() function to return the correct item
+    newPlugin.item = function (index) {
+      return newPlugin[index]
+    }
+
+    return newPlugin
+  }
+
+  // We need the original length so we can reference it (as we will change it)
+  const plugins = window.navigator.plugins
+  const originalPluginsLength = plugins.length
+  const originalItemFunction = plugins.item
+
+  // Adds a fake plugin for the given index on fakePluginData
+  const addPluginAtIndex = (newPlugin, index) => {
+    const pluginPosition = originalPluginsLength + index
+    window.navigator.plugins[pluginPosition] = newPlugin
+    window.navigator.plugins[newPlugin.name] = newPlugin
+  }
+
+  for (const [index, pluginData] of fakePluginData.entries()) {
+    const newPlugin = makeFakePlugin(pluginData)
+    addPluginAtIndex(newPlugin, index)
+  }
+
+  // Adjust the length of the original plugin array
+  Reflect.defineProperty(window.navigator.plugins, 'length', {
+    get: function () {
+      return originalPluginsLength + fakePluginData.length
+    }
+  })
+
+  // Fix .item() function to return the correct item
+  window.PluginArray.prototype.item = function (index) {
+    if (index < originalPluginsLength) {
+      return Reflect.apply(originalItemFunction, plugins, arguments)
+    } else {
+      return plugins[index]
+    }
+  }
+
+  // 4. Farble speech synthesizer
+  const makeFakeVoiceFromVoice = (voice) => {
+    const voicePrototype = Object.getPrototypeOf(voice)
+
+    const newVoice = Object.create(voicePrototype, {
+      name: {
+        get: function () {
+          return fakeVoiceName
+        }
+      },
+      voiceURI: {
+        get: function () {
+          return voice.voiceURI
+        }
+      },
+      lang: {
+        get: function () {
+          return voice.lang
+        }
+      },
+      localService: {
+        get: function () {
+          return voice.localService
+        }
+      },
+      default: {
+        get: function () {
+          return false
+        }
+      }
+    })
+
+    return newVoice
+  }
+
+  let originalVoice
+  let fakeVoice
+  let passedFakeVoice
+
+  // We need to override the voice property to allow our fake voice to work
+  const descriptor = Reflect.getOwnPropertyDescriptor(SpeechSynthesisUtterance.prototype, 'voice')
+  Reflect.defineProperty(SpeechSynthesisUtterance.prototype, 'voice', {
+    get: function () {
+      if (passedFakeVoice === undefined) {
```

Comment:
`if (!passedFakeVoice)` because it can possibly be null.

Reply:
I actually do a `passedFakeVoice = undefined` so that should work. But I'm not totally sure if this is a good way of doing things. Suggestions are more than welcome. I want to do things the proper way.